crates/but/src/branch/mod.rs

```diff
@@ -14,21 +16,88 @@ pub enum Subcommands {
-            Ok(())
+        Subcommands::New {
+            branch_name,
+            anchor,
+        } => {
+            if let Some(anchor_str) = anchor {
+                // Use the new create_reference API when anchor is provided
+                let ctx = CommandContext::open(
+                    &project,
+                    AppSettings::load_from_default_path_creating()?,
+                )?;
+                let mut ctx = ctx; // Make mutable for CliId resolution
+
+                // Resolve the anchor string to a CliId
+                let anchor_ids = crate::id::CliId::from_str(&mut ctx, anchor_str)?;
+                if anchor_ids.is_empty() {
+                    return Err(anyhow::anyhow!("Could not find anchor: {}", anchor_str));
+                }
+                if anchor_ids.len() > 1 {
+                    return Err(anyhow::anyhow!(
+                        "Ambiguous anchor '{}', matches multiple items",
+                        anchor_str
+                    ));
+                }
+                let anchor_id = &anchor_ids[0];
+
+                // Get branch name or use canned name
+                let branch_name = if let Some(name) = branch_name {
+                    name.clone()
+                } else {
+                    but_api::workspace::canned_branch_name(project.id)?
+                };
+
+                // Create the anchor for create_reference
+                let anchor = match anchor_id {
+                    crate::id::CliId::Commit { oid } => {
+                        Some(but_api::stack::create_reference::Anchor::AtCommit {
+                            commit_id: (*oid).into(),
+                            position: but_workspace::branch::create_reference::Position::Above,
+                        })
+                    }
+                    crate::id::CliId::Branch { name } => {
+                        Some(but_api::stack::create_reference::Anchor::AtReference {
+                            short_name: name.clone(),
+                            position: but_workspace::branch::create_reference::Position::Above,
+                        })
+                    }
+                    _ => {
+                        return Err(anyhow::anyhow!(
+                            "Invalid anchor type: {}, expected commit or branch",
+                            anchor_id.kind()
+                        ));
+                    }
+                };
+
+                // Use create_reference API
+                let request = but_api::stack::create_reference::Request {
+                    new_name: branch_name.clone(),
+                    anchor,
+                };
+                but_api::stack::create_reference(project.id, request)?;
+                println!("Created branch {branch_name}");
+                Ok(())
+            } else {
+                // Create an independent branch
+                let req = gitbutler_branch::BranchCreateRequest {
```

Comment:
That's interesting, I thought `but_api::create_branch()` does the same, but using the new API.

https://github.com/gitbutlerapp/gitbutler/blob/29a0d937602e6f2523011a861a1f5fafbf3f3a69/crates/but-api/src/commands/stack.rs#L96

That's what the UI calls.

Reply:
Hmm the frontend still does the `create_virtual_branch` command so this just does the same
Maybe it should be updated in both places....
<img width="1434" height="961" alt="image" src="https://github.com/user-attachments/assets/5fa919b3-6475-472a-b986-979fccd3a9a4" />

Reply:
Ah, right, I was confused. Somehow I thought there would only be one function for that, but in fact there are two, this one and the one I mentioned, and *both* are used by the UI for some reason :). Both support `ws3` and will call the respective new function, so there is no issue with it - probably this one has better 

This is probably where one would apply another fix then.

I also think what I was responding to here is to generally avoid calling anything with `virtual` in it 😅.

Reply:
I opened https://github.com/gitbutlerapp/gitbutler/pull/10492 for this, with the kind request to validate it as I didn't do that